gems/sorbet-runtime/lib/types/private/methods/_methods.rb

```diff
@@ -221,7 +232,7 @@ def self._on_method_added(hook_mod, method_name, is_singleton_method: false)
     new_method = mod.instance_method(method_name)
     key = method_to_key(new_method)
     @sig_wrappers[key] = sig_block
-    if current_declaration.final
+    if current_declaration.final || current_mod_final
```

Comment:
Things like this make me increasingly convinced that someone's going to forget to check `|| current_mod_final` somewhere (either in the runtime or statically).